Crash in OGR2SQLITE_static_register

## Expected behavior and actual behavior.

when loading project on iPad (see https://github.com/lutraconsulting/input/issues/435) it crashes ogr in `pApi->create_module`. Probably due to static build (iOS does not support dynamic libraries), argument _pApi is nullptr and hence `pApi->create_module` triggers `EXC_BAD_ACCESS` see https://github.com/OSGeo/gdal/blob/6c20fa95d33fd0452329067f766b2d71a5ebd87f/gdal/ogr/ogrsf_frmts/sqlite/ogrsqlitevirtualogr.cpp#L2593

also during compilation of the ogrsqlitevirtualogr.cpp file there is a warning

```
ogrsqlitevirtualogr.cpp:74:5: warning: 'sqlite3_auto_extension' is deprecated: first deprecated in iOS 8.2 - Process-global auto extensions are not supported on Apple platforms
      [-Wdeprecated-declarations]
    sqlite3_auto_extension ((void (*)(void)) OGR2SQLITE_static_register);
    ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.4.sdk/usr/include/sqlite3.h:5997:16: note: 'sqlite3_auto_extension' has been
      explicitly marked deprecated here
SQLITE_API int sqlite3_auto_extension(void(*xEntryPoint)(void));
```

## proposed solution

the crash goes away if I change the line to 

` if( !pApi || pApi->create_module == nullptr ) `

or just remove the call to `sqlite3_auto_extension` in `OGR2SQLITE_Register` alltogether.

I am not sure if we have some flag for IOS builds to ifdef it? 

## Operating system

iOS 12.0

## GDAL version and provenance

GDAL 2.4.0 custom STATIC build from [input-sdk](https://github.com/lutraconsulting/input-sdk/blob/master/ios/recipes/gdal/recipe.sh) with system SQLITE3 (3.24.0)

--- a/gdal/ogr/ogrsf_frmts/sqlite/ogrsqlitevirtualogr.cpp
+++ b/gdal/ogr/ogrsf_frmts/sqlite/ogrsqlitevirtualogr.cpp
@@ -2590,7 +2590,7 @@ int OGR2SQLITE_static_register (sqlite3 * hDB, char **pzErrMsg, void * _pApi)
 {
     const sqlite3_api_routines * pApi = (const sqlite3_api_routines * )_pApi;
 #ifndef WIN32
-    if( pApi->create_module == nullptr )
+    if( ( pApi == nullptr ) || ( pApi->create_module == nullptr ) )
     {
         pApi = &OGRSQLITE_static_routines;
     }

